src/app/application.cpp

```diff
@@ -888,9 +891,19 @@ int Application::exec()
 #endif // DISABLE_GUI
 
 #ifndef DISABLE_WEBUI
+#ifndef DISABLE_GUI
         m_webui = new WebUI(this);
-#ifdef DISABLE_GUI
-        connect(m_webui, &WebUI::error, this, [](const QString &message) { fprintf(stderr, "%s\n", qUtf8Printable(message)); });
+#else
+        const auto *pref = Preferences::instance();
+
+        const QString tempPassword = pref->getWebUIPassword().isEmpty()
+                ? u"%1"_s.arg(QString::number(Utils::Random::rand(0, 999999999)), 9, u'0')
```

Comment:
1. The arg() signature is wrong. As @Chocobo1 had mentioned it should be:
```c++
arg(QString::number(Utils::Random::rand(0, 999999999)), 9, 10, u'0')
```

2. Secondly, even for a temporary password, I don't consider the generated one as sufficiently secure based on this method. I would propose something like this:
```c++
QString generatePasswd()
{
  QString pass;
  pass.reserve(9);
  while (pass.length() < 9)
  {
    // ASCII min/max
    const auto num = Utils::Random::rand(48, 122);
    // ASCII 96 is a somewhat difficult character to write via keyboard
    if (num == 96) continue;

    pass.append(static_cast<char16_t>(num));
  }
}
```

Reply:
> 1. The arg() signature is wrong. As @Chocobo1 had mentioned it should be:
> 
> ```c++
> arg(QString::number(Utils::Random::rand(0, 999999999)), 9, 10, u'0')
> ```

You got it wrong. I just chose another valid overload that has less ambiguous in recognizing parameter types.
Your example is wrong.

Reply:
> 2\. Secondly, even for a temporary password, I don't consider the generated one as sufficiently secure based on this method

I just took advantage of what was suggested in the early discussions.
>I would propose something like this:

I can do that if @Chocobo1 also approve it. I don't want to get stuck here.

Reply:
>You got it wrong. I just chose another valid overload that has less ambiguous in recognizing parameter types.
Your example is wrong.

+1
QString::number() returns a QString and I was looking at the wrong overload signatures.

Reply:
>Secondly, even for a temporary password, I don't consider the generated one as sufficiently secure based on this method. I would propose something like this:

Sure it is harder to brute force but it also might be a bit troublesome for users to input. Regarding to brute force, webui already has mechanism to filter that so I won't worry about this attack.
I don't have strong preference here.

Reply:
>Sure it is harder to brute force but it also might be a bit troublesome for users to input.

Then it could be something like:
```c++
QString generatePasswd()
{
  static const QString alphanum = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
  QString pass;
  pass.reserve(9);
  while (pass.length() < 9)
  {
    const auto num = Utils::Random::rand(0, (alphanum.size() - 1));
    pass.append(alphanum[num]);
  }

  return pass;
}
```

IMO, having a numeric only pass is not good enough for something exposed to the Internet.

Reply:
>IMO, having a numeric only password is not good enough for something exposed to the Internet.

OK, I don't mind it.

Reply:
Done.